source/common/network/address_impl.h

```diff
@@ -103,8 +103,9 @@ class Ipv4Instance : public InstanceBase {
     const std::string& addressAsString() const override { return friendly_address_; }
     bool isAnyAddress() const override { return ipv4_.address_.sin_addr.s_addr == INADDR_ANY; }
     bool isUnicastAddress() const override {
-      return !isAnyAddress() && ipv4_.address_.sin_addr.s_addr != INADDR_BROADCAST &&
-             !IN_MULTICAST(ipv4_.address_.sin_addr.s_addr);
+      return !isAnyAddress() && (ipv4_.address_.sin_addr.s_addr != INADDR_BROADCAST) &&
+             // inlined IN_MULTICAST() to avoid byte swapping
```

Comment:
For my understanding why is this preferred over `IN_MULTICAST(htonl(ipv4_.address_.sin_addr.s_addr))` (or something similar)?